IaC/scripts/create-cluster-env.bash

```diff
@@ -0,0 +1,398 @@
+echo -e "\nInstall the Istio Operator and Components on AKS"
+istioctl operator init
+kubectl create namespace istio-system
+kubectl apply -f $REPO_ROOT/IaC/AKS/cluster/manifests/istio/istio.aks.yaml
+# One-line sleep-waiting on deployment deployment/istio-ingressgateway
+echo -e "\nWaiting for Istio deployment..."
+while ! kubectl wait --for=condition=available deployment/istio-ingressgateway -n istio-system ;do sleep 1;done
+# Validate the Istio installation
+kubectl get all --namespace istio-system
+
+# Enable automatic sidecar injection in the ngsa namespace:
+kubectl create namespace ngsa
+kubectl label namespace ngsa istio-injection=enabled
+
+INGRESS_PIP=$(kubectl --namespace istio-system get svc -l istio=ingressgateway -o jsonpath='{.items[0].status.loadBalancer.ingress[0].ip}')
+echo -e "\nIngree PIP: ${INGRESS_PIP}"
+# Install KEDA
+KEDA_VERSION=2.0.0
+kubectl create namespace keda
+helm install keda kedacore/keda --namespace keda --version $KEDA_VERSION --atomic
+
+# Create Secrets for NGSA
+Ngsa_Log_Analytics_Wid=$(az monitor log-analytics workspace show -g $Ngsa_Log_Analytics_RG -n $Ngsa_Log_Analytics_Name --query customerId -o tsv ${subs_arg})
+Ngsa_Log_Analytics_SharedKey=$(az monitor log-analytics workspace get-shared-keys -g $Ngsa_Log_Analytics_RG -n $Ngsa_Log_Analytics_Name --query primarySharedKey -o tsv ${subs_arg})
+kubectl create secret generic ngsa-secrets --namespace ngsa --from-literal=CosmosDatabase=$Imdb_DB --from-literal=CosmosCollection=$Imdb_Col --from-literal=CosmosKey=${Ngsa_Imdb_CM_Key} --from-literal=CosmosUrl=${Ngsa_Imdb_CM_Url}
+
+# Setup SSL/DNS with a registered domain if available
+# If Domain name is not given, use the default one
+if [[ -z ${Ngsa_Domain_Name} || ${Ngsa_Domain_Name} == 'nip.io' ]]; then
+    Ngsa_Domain_Name=nip.io
+    # set application endpoint
+    export Ngsa_App_Endpoint="${INGRESS_PIP}.${Ngsa_Domain_Name}"
+    cert_enabled=false
+    export Ngsa_Https_App_Endpoint="http://${Ngsa_App_Endpoint}"
+else
+    export Ngsa_Email=$Ngsa_Email
+    [[ -z ${Ngsa_DNS_RG} ]] && Ngsa_DNS_RG=dns-rg
+    # Check if DNS resource group exists
+    if [[ "$(az group exists -n $Ngsa_DNS_RG ${subs_arg})" == "false" ]]; then
+        # If false, create DNS resource group
+        aze group create -n $Ngsa_DNS_RG -l $Ngsa_Location -o table ${subs_arg}
+    fi
+
+    # Check if DNS Zone exists
+    if ! az network dns zone show --name $Ngsa_Domain_Name -g $Ngsa_DNS_RG -o table ${subs_arg}; then
+        # If not found, create the DNS Zone.
+        aze network dns zone create -g $Ngsa_DNS_RG -n $Ngsa_Domain_Name -o table ${subs_arg}
+    fi
+    # Add DNS A record for the Istio ingress gateway.
+    aze network dns record-set a add-record -g $Ngsa_DNS_RG -z $Ngsa_Domain_Name -n $Ngsa_Name -a $INGRESS_PIP -o table ${subs_arg}
+
+    # Show the Azure nameservers for your DNS Zone.
+    aze network dns zone show -n $Ngsa_Domain_Name -g $Ngsa_DNS_RG --query nameServers -o tsv ${subs_arg}
```

Comment:
Lines 326, 332, 335, and 338, should those be "az network ..." instead of "aze network ..."?